browser/mac/sparkle_glue.mm

```diff
@@ -45,69 +48,23 @@
       [NSString stringWithFormat:@"AppcastItem(Date: %@, Version: %@)",
           [item performSelector:@selector(dateString)],
           GetVersionFromAppcastItem(item)];
-  return [description UTF8String];
+  return base::SysNSStringToUTF8(description);
 }
 
-// Adaptor for scheduling an Objective-C method call in TaskScheduler.
-class PerformBridge : public base::RefCountedThreadSafe<PerformBridge> {
- public:
-
-  // Call |sel| on |target| with |arg| in a WorkerPool thread.
-  // |target| and |arg| are retained, |arg| may be |nil|.
-  static void PostPerform(id target, SEL sel, id arg) {
-    DCHECK(target);
-    DCHECK(sel);
-
-    scoped_refptr<PerformBridge> op = new PerformBridge(target, sel, arg);
-    base::ThreadPool::PostTask(
-        FROM_HERE,
-        {base::MayBlock(), base::TaskPriority::BEST_EFFORT,
-         base::TaskShutdownBehavior::CONTINUE_ON_SHUTDOWN},
-        base::BindOnce(&PerformBridge::Run, std::move(op)));
-  }
-
-  // Convenience for the no-argument case.
-  static void PostPerform(id target, SEL sel) {
-    PostPerform(target, sel, nil);
-  }
-
- private:
-  // Allow RefCountedThreadSafe<> to delete.
-  friend class base::RefCountedThreadSafe<PerformBridge>;
-
-  PerformBridge(id target, SEL sel, id arg)
-      : target_([target retain]),
-        sel_(sel),
-        arg_([arg retain]) {
-  }
-
-  ~PerformBridge() {}
-
-  // Happens on a WorkerPool thread.
-  void Run() {
-    base::mac::ScopedNSAutoreleasePool pool;
-    [target_ performSelector:sel_ withObject:arg_];
-  }
-
-  base::scoped_nsobject<id> target_;
-  SEL sel_;
-  base::scoped_nsobject<id> arg_;
-};
-
 }  // namespace
 
 @implementation SparkleGlue
 {
-  SUUpdater* su_updater_;
+  SUUpdater* _su_updater;
```

Comment:
I think this needs `__weak` as shared pointer is assigned to it.  (`_su_updater = [SUUpdater sharedUpdater]`).
`SUUpdater* __weak _su_updater;`

Reply:
Thanks again @simonhong! Updated as you suggest.

Reply:
Sorry @mkarolin , I think this also needs `__strong` reference as  it's referenced as long as `SparkleGlue` is live.

Reply:
No problem 👍 Updated in https://github.com/brave/brave-core/pull/18808/commits/379d3902780ae5f4e76901b9f8ec1a13c843924c